### PR TITLE
Fix log handling

### PR DESCRIPTION
**Pull request by cortadocodes**
### Enhancements
- [x] Improve `Service` log messages
- [x] Keep full module path in logging context when using analysis logger 
- [x] Demote storage client log messages to `DEBUG` level

### Fixes
- [x] Fix and improve `apply_log_handler` function
- [x] Apply log handler to the root logger in top-level `__init__` module
- [x] Stop duplication of timestamp in Google Cloud Run logs by applying different handler/formatter if `PLATFORM` environment variable is present and equal to `GOOGLE_CLOUD_RUN`
- [x] Add `PLATFORM=GOOGLE_CLOUD_RUN` environment variable to docker image 
- [x] Remove applying of log handler from incorrect places 
- [x] Remove extra logger in `runner` module
- [x] Stop analysis logger propagating to the root logger and repeating log messages 

### Reversions
- [x] Remove `show-twined-logs` option from CLI

### Testing
- [x] Silence GCP Storage emulator logs below `ERROR` level
- [x] Ensure unmocked `Subscription` test always deletes topic and subscription
- [x] Remove extra logger from `tests.base`
- [x] Avoid irrelevant exception in Cloud Run tests 

### Operations
- [x] Remove unnecessary preceding underscores from ARGs in Dockerfile

### CODE REVIEW

File: octue/__init__.py [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
+from .logging_handlers import FORMATTER_WITHOUT_TIMESTAMP, apply_log_handler
 from .runner import Runner
 
 
-__all__ = "LOG_FORMAT", "Runner"
-package_logger = logging.getLogger(__name__)
+__all__ = ("Runner",)
+
+
+if os.environ.get("PLATFORM") == "GOOGLE_CLOUD_RUN":
```
Comment (thclark) [inline octue/__init__.py:10]: Is this an environment variable that we have to set now? Just wondering if this should be a function to detect where we're running. And what if PLATFORM not set, shouldn't you be providing a default?

If we have control over the env var name, we should call it something like COMPUTE_PROVIDER to be systematic with the language we've used elsewhere.

File: octue/__init__.py [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
+    # Use a log handler with a formatter that doesn't include the timestamp in the log message context to avoid the date
+    # appearing twice in the Google Cloud Run logs (Google adds its own timestamp to log messages).
+    logger = apply_log_handler(__name__, formatter=FORMATTER_WITHOUT_TIMESTAMP)
+else:
+    apply_log_handler()
```
Comment (thclark) [inline octue/__init__.py:15]: why is this called without __name__? Was expecting a call to be similar other than the override of the handler...

File: octue/cloud/deployment/google/cloud_run.py [hunk 3]
```diff
@@ -6,21 +6,16 @@
...
 logger = logging.getLogger(__name__)
-apply_log_handler(logger, log_level=logging.INFO)
-
+app = Flask(__name__)
 
 DEPLOYMENT_CONFIGURATION_PATH = "deployment_configuration.json"
 
 
-app = Flask(__name__)
-
```
Comment (thclark) [inline octue/cloud/deployment/google/cloud_run.py:22]: Why has this switched? Better to keep constants sat at the top by convention, no?

File: octue/mixins/loggable.py [hunk 4]
```diff
@@ -8,22 +8,29 @@ class Loggable:
     same logger instance; this can be confusing if you overload __init_logger__ in multiple different ways.
 
     ```
-    class MyResource(Logged):
+    class MyResource(Loggable):
```
Comment (thclark) [inline octue/mixins/loggable.py:11]: Nice catch

File: tests/cloud/deployment/google/test_cloud_run.py [hunk 5]
```diff
@@ -78,12 +78,13 @@ def test_with_no_deployment_configuration_file(self):
         with mock.patch.dict(os.environ, {"SERVICE_ID": self.EXAMPLE_SERVICE_ID}):
             with mock.patch("octue.cloud.deployment.google.cloud_run.Runner") as mock_runner:
                 with mock.patch("octue.cloud.pub_sub.service.Topic", new=MockTopic):
-                    cloud_run.answer_question(
-                        project_name="a-project-name",
-                        data={},
-                        question_uuid="8c859f87-b594-4297-883f-cd1c7718ef29",
-                        credentials_environment_variable="GOOGLE_APPLICATION_CREDENTIALS",
-                    )
+                    with mock.patch("octue.cloud.deployment.google.cloud_run.Service"):
```
Comment (thclark) [inline tests/cloud/deployment/google/test_cloud_run.py:81]: Next time you touch this, suggest creating a reusable context manager that patches everything as necessary.

File: tests/test_cli.py [hunk 6]
```diff
@@ -60,57 +60,6 @@ def test_run_command_works_with_data_dir(self):
...
-                        f'--data-dir={os.path.join(TESTS_DIR, "data", "data_dir_with_no_manifests")}',
-                        f"--output-dir={temporary_directory}",
-                    ],
-                )
-
-            mock_log_handler_emit.assert_called()
-
-            self.assertTrue(
-                any(
-                    "Showing package logs as well as analysis logs (the package logs are recommended for software "
```
Comment (thclark) [inline tests/test_cli.py:83]: So this is what I meant about some logs going to the cloud, some logs from the actual package going back to the asker. Will this capability be brought back in?

**Review by thclark — COMMENTED** [review]
Minor questions mostly

File: octue/__init__.py (hunk 1 above)
Comment (cortadocodes) [reply octue/__init__.py:10]: We don't have to set it anywhere but Google Cloud Run. I left the default as `None` but I could set it as `LOCAL` but that might not always be correct. We do have control over the name of the envvar - shall I change it to that?

File: octue/__init__.py (hunk 2 above)
Comment (cortadocodes) [reply octue/__init__.py:15]: It doesn't make a difference whether it's given `__name__` or `None` so I'll give it `__name__` for symmetry

File: octue/cloud/deployment/google/cloud_run.py (hunk 3 above)
Comment (cortadocodes) [reply octue/cloud/deployment/google/cloud_run.py:22]: I wasn't sure whether it's better to have constants first (e.g. `DEPLOYMENT_CONFIGURATION_PATH`) and global objects after, or the reverse. What do you think?

File: tests/test_cli.py (hunk 6 above)
Comment (cortadocodes) [reply tests/test_cli.py:83]: I think the capability as it was there didn't really work as intended in the end and might have been misguided by me as a way to get around the bugs resolved in this PR that I didn't know existed at the time. I was planning to start from scratch on streaming some logs from the cloud etc. but in a different PR. I can do it in this one if you prefer?

Comment (thclark) [reply tests/test_cli.py:83]: Happy for it to be separate

File: octue/cloud/deployment/google/cloud_run.py (hunk 3 above)
Comment (thclark) [reply octue/cloud/deployment/google/cloud_run.py:22]: I'd always expect to see constants first, by generalish convention in most langauses (except for some with explicit and weird globals like fortran's shared modules stuff which had to be defined at the top)

File: octue/__init__.py (hunk 1 above)
Comment (thclark) [reply octue/__init__.py:10]: I'd maybe go with a default of UNKNOWN?

File: octue/__init__.py (hunk 2 above)
Comment (cortadocodes) [reply octue/__init__.py]: Update: it matters a lot whether we use `__name__` or `None`/nothing for the argument to `apply_log_handler`. If we use `__name__`, the handler isn't applied to the root logger so we get every log message twice - once formatted (`__name__` logger) and once unformatted (root logger). Using `None` applies the handler to the root logger; all other loggers propagate messages up to it, so messages are only printed once and in the correct format.

@thclark

**Pull request merged**